### PR TITLE
Add support for getting Droplets by name

### LINKED ISSUE #895: `doctl compute droplet get <droplet-id|droplet-name>`
**What is the problem this feature would solve? Please describe.**

`doctl compute droplet get` supports only droplet ID whereas `doctl compute droplet delete` supports ID and name

```console
doctl compute droplet get --help

Usage:
  doctl compute droplet get <droplet-id> [flags]

doctl compute droplet delete --help

Usage:
  doctl compute droplet delete <droplet-id|droplet-name>... [flags]
```

**Describe the solution you'd like**

It would be helpful to `doctl compute droplet get <droplet-name>` to save `doctl compute droplet list` and then either `grep`ping the ID  or `--output=JSON` and then `jq`ing it, in order to get values from a specific droplet's configuration.

Or, I'm missing something!?

### PR DESCRIPTION
**Pull request by mamercad**
Might fix #895 -- I'm not very experienced in Golang.

```
❯ ./builds/doctl compute droplet list
ID           Name             Public IPv4       Private IPv4    Public IPv6    Memory    VCPUs    Disk    Region    Image                     VPC UUID                                Status    Tags    Features              Volumes
213635352    hacktoberfest    142.93.240.125    10.116.0.2                     1024      1        25      nyc1      Ubuntu 20.04 (LTS) x64    0c20a2e6-b9ca-43fd-8b22-def8741321ac    active            private_networking    
213653028    hacktoberfest    142.93.5.117      10.116.0.3                     1024      1        25      nyc1      Ubuntu 20.04 (LTS) x64    0c20a2e6-b9ca-43fd-8b22-def8741321ac    active            private_networking
```

```
❯ ./builds/doctl compute droplet get hacktoberfest
Error: Multiple Droplets with the name "hacktoberfest" found.
```

```
❯ ./builds/doctl compute droplet delete 213653028
Warning: Are you sure you want to delete this Droplet? (y/N) ? y
```

New functionality:
```
❯ ./builds/doctl compute droplet get hacktoberfest
ID           Name             Public IPv4       Private IPv4    Public IPv6    Memory    VCPUs    Disk    Region    Image                     VPC UUID                                Status    Tags    Features              Volumes
213635352    hacktoberfest    142.93.240.125    10.116.0.2                     1024      1        25      nyc1      Ubuntu 20.04 (LTS) x64    0c20a2e6-b9ca-43fd-8b22-def8741321ac    active            private_networking
```

```
❯ ./builds/doctl compute droplet delete 213635352
Warning: Are you sure you want to delete this Droplet? (y/N) ? y
```

```
❯ ./builds/doctl compute droplet get hacktoberfest
Error: Droplet with the name "hacktoberfest" could not be found.
```

### CODE REVIEW

**Review by andrewsomething — CHANGES_REQUESTED** [review]
Thanks for taking this on @mamercad!

In order to reduce some duplication, I think it would be best to reuse the existing `matchDroplets` function that is called when deleting by name.

https://github.com/digitalocean/doctl/blob/master/commands/droplets.go#L516

You can see how it gets used by the delete command here:

https://github.com/digitalocean/doctl/blob/master/commands/droplets.go#L501-L509

**Comment by mamercad** [comment]
This isn't right, still working on it.

**Comment by mamercad** [comment]
```
❯ ./builds/doctl compute droplet list
ID           Name             Public IPv4        Private IPv4    Public IPv6    Memory    VCPUs    Disk    Region    Image                     VPC UUID                                Status    Tags    Features              Volumes
213989132    hacktoberfest    67.205.153.187     10.116.0.2                     1024      1        25      nyc1      Ubuntu 20.04 (LTS) x64    0c20a2e6-b9ca-43fd-8b22-def8741321ac    active            private_networking    
214047592    hacktoberfest    167.172.148.146    10.116.0.3                     1024      1        25      nyc1      Ubuntu 20.04 (LTS) x64    0c20a2e6-b9ca-43fd-8b22-def8741321ac    active            private_networking    
214049768    test             157.230.215.122    10.116.0.4                     1024      1        25      nyc1      Ubuntu 20.04 (LTS) x64    0c20a2e6-b9ca-43fd-8b22-def8741321ac    active            private_networking    

❯ ./builds/doctl compute droplet get hacktoberfest
Error: There are 2 Droplets with the name "hacktoberfest"; please provide a specific Droplet ID. [213989132, 214047592]

❯ ./builds/doctl compute droplet get test
ID           Name    Public IPv4        Private IPv4    Public IPv6    Memory    VCPUs    Disk    Region    Image                     VPC UUID                                Status    Tags    Features              Volumes
214049768    test    157.230.215.122    10.116.0.4                     1024      1        25      nyc1      Ubuntu 20.04 (LTS) x64    0c20a2e6-b9ca-43fd-8b22-def8741321ac    active            private_networking
```

Seems to work now.

File: commands/droplets.go [hunk 1]
```diff
@@ -556,9 +556,35 @@ func matchDroplets(ids []string, ds do.DropletsService, fn matchDropletsFn) erro
 
 // RunDropletGet returns a droplet.
 func RunDropletGet(c *CmdConfig) error {
+
 	id, err := getDropletIDArg(c.NS, c.Args)
 	if err != nil {
```
Comment (andrewsomething) [inline commands/droplets.go:561]: This check can be removed. It always results in an error when passing a string. `getDropletIDArg` ensures that only one argument is provided and that it is an integer. Now that integers and strings are valid arguments, the check you added below will be enough to validate input.

File: commands/droplets.go [hunk 2]
```diff
@@ -556,9 +556,35 @@ func matchDroplets(ids []string, ds do.DropletsService, fn matchDropletsFn) erro
...
-		return err
+		// Try to find Droplet by name
+		if len(c.Args) < 1 {
```
Comment (andrewsomething) [inline commands/droplets.go:563]:
```suggestion
		if len(c.Args) != 1 {
```

The get command only returns one result, so should only accept one argument.

File: commands/droplets.go [hunk 3]
```diff
@@ -556,9 +556,35 @@ func matchDroplets(ids []string, ds do.DropletsService, fn matchDropletsFn) erro
...
+			if len(matches) == 1 {
+				id = matches[0]
+			}
+			return nil
+		}
+
+		err := matchDroplets(c.Args, ds, fn)
+		if err != nil {
+			return err
+		}
```
Comment (andrewsomething) [inline commands/droplets.go:587]: Some of this is still duplicating some of what `matchDroplets` provides. The "anonymous" function here that is passed to `matchDroplets`, is called with a list of IDs which have already had the conversion from name to ID. This also duplicates the `ds.Get(id)` further below, adding another network request.

It's probably a bit cleaner to move the display logic into the function and just return the call to `matchDroplets` at the end.

```go
	fn := func(ids []int) error {
		for _, id := range ids {
			d, err := ds.Get(id)
			if err != nil {
				return err
			}

			item := &displayers.Droplet{Droplets: do.Droplets{*d}}
			if getTemplate != "" {
				t := template.New("Get template")
				t, err = t.Parse(getTemplate)
				if err != nil {
					return err
				}
				return t.Execute(c.Out, d)
			}
			return c.Display(item)
		}
		return nil
	}

	return matchDroplets(c.Args, ds, fn)
```

**Review by andrewsomething — CHANGES_REQUESTED** [review]

**Comment by mamercad** [comment]
```
❯ ./builds/doctl compute droplet list | grep test
214119531    test1                               64.227.3.17        10.116.0.6                     1024      1        25      nyc1      Ubuntu 20.04 (LTS) x64        0c20a2e6-b9ca-43fd-8b22-def8741321ac    active                                                               private_networking    
214119620    test1                               134.122.117.15     10.116.0.7                     1024      1        25      nyc1      Ubuntu 20.04 (LTS) x64        0c20a2e6-b9ca-43fd-8b22-def8741321ac    active                                                               private_networking    
214119947    test2                               157.230.232.122    10.116.0.8                     1024      1        25      nyc1      Ubuntu 20.04 (LTS) x64        0c20a2e6-b9ca-43fd-8b22-def8741321ac    new                                                                  private_networking    

❯ ./builds/doctl compute droplet get test2
ID           Name     Public IPv4        Private IPv4    Public IPv6    Memory    VCPUs    Disk    Region    Image                     VPC UUID                                Status    Tags    Features              Volumes
214119947    test2    157.230.232.122    10.116.0.8                     1024      1        25      nyc1      Ubuntu 20.04 (LTS) x64    0c20a2e6-b9ca-43fd-8b22-def8741321ac    active            private_networking    

❯ ./builds/doctl compute droplet get test1
Error: There are 2 Droplets with the name "test1"; please provide a specific Droplet ID. [214119531, 214119620]

❯ ./builds/doctl compute droplet get 214119531
ID           Name     Public IPv4    Private IPv4    Public IPv6    Memory    VCPUs    Disk    Region    Image                     VPC UUID                                Status    Tags    Features              Volumes
214119531    test1    64.227.3.17    10.116.0.6                     1024      1        25      nyc1      Ubuntu 20.04 (LTS) x64    0c20a2e6-b9ca-43fd-8b22-def8741321ac    active            private_networking
```

**Review by mamercad — COMMENTED** [review]
Ah, thanks for helping me, again!

**Review by andrewsomething — APPROVED** [review]
:+1:  Thanks again for this! Just waiting for Travis to go green (queue times have been very long recently).

**Pull request merged**